plugin/rewrite/reverter.go

```diff
@@ -23,15 +25,16 @@ type ResponseRule struct {
 type ResponseReverter struct {
 	dns.ResponseWriter
 	originalQuestion dns.Question
-	ResponseRewrite  bool
 	ResponseRules    []ResponseRule
+	noSave           bool
 }
 
 // NewResponseReverter returns a pointer to a new ResponseReverter.
-func NewResponseReverter(w dns.ResponseWriter, r *dns.Msg) *ResponseReverter {
+func NewResponseReverter(w dns.ResponseWriter, r *dns.Msg, noSaveQuestion bool) *ResponseReverter {
 	return &ResponseReverter{
 		ResponseWriter:   w,
 		originalQuestion: r.Question[0],
+		noSave:           noSaveQuestion,
```

Comment:
```suggestion
		restore:           restoreQuestion,
```

Reply:
Better world be `noRestore`, the default (false) should be the standard behavior, which is restoring the original question in the response handling. For testing purposes this will then be set to true.